### PR TITLE
Feature/pools table

### PR DESCRIPTION
**Pull request by Chef-Cheems**
Pools table
Preview - https://awesome-raman-0813c6.netlify.app/

### CODE REVIEW

**Review by Chef-Cheems — COMMENTED** [review]
Basically table consists of main table wrapper `<PoolsTable>` which has bunch of `<PoolRow>` inside, each row consist of cells present in `Cells` directory. The data calculation logic is pretty much the same as in cards with the exception that vault is not separated to its own "VaultRow" component but integrated into normal flow (i.e. there is bunch of `if (isAutoVault) ...` conditions here.

File: src/components/Balance.tsx
Comment (hachiojidev) [inline src/components/Balance.tsx:11]: I assume this is necessary but it shouldn't be. `onClick` should be inherited from `TextProps`

File: src/state/pools/index.ts
Comment (hachiojidev) [reply src/state/pools/index.ts:18]: Not a priority but we have used `isInitialized` throughout the app for this.

File: src/views/Pools/components/CakeVaultCard/UnstakingFeeCountdownRow.tsx
Comment (hachiojidev) [inline src/views/Pools/components/CakeVaultCard/UnstakingFeeCountdownRow.tsx:10]: Please prefix this with `is` or `has`

File: src/views/Pools/components/HelpButton.tsx
Comment (hachiojidev) [inline src/views/Pools/components/HelpButton.tsx:29]: You can't use `endIcon` here?

File: src/views/Pools/components/PoolTabButtons.tsx
Comment (hachiojidev) [inline src/views/Pools/components/PoolTabButtons.tsx:58]: Just an FYI, I see in other spots you are using `null`. Using `null` is preferable as it doesn't create a new style.

File: src/views/Pools/components/PoolsTable/ActionPanel/ActionPanel.tsx
Comment (hachiojidev) [inline src/views/Pools/components/PoolsTable/ActionPanel/ActionPanel.tsx:88]: This is not exported from the toolkit? If not, then can you add a TODO here to do so.

File: src/views/Pools/components/PoolsTable/Apr.tsx
Comment (hachiojidev) [inline src/views/Pools/components/PoolsTable/Apr.tsx:42]: `288` should be in a config I think.

File: src/views/Pools/components/PoolsTable/Cells/EarningsCell.tsx
Comment (hachiojidev) [inline src/views/Pools/components/PoolsTable/Cells/EarningsCell.tsx:24]: Why 4.5?

File: src/views/Pools/components/PoolsTable/PoolsTable.tsx
Comment (hachiojidev) [inline src/views/Pools/components/PoolsTable/PoolsTable.tsx:20]: Do we need to add this to the UI Kit? If so please add a TODO

File: src/state/pools/index.ts
Comment (Chef-Cheems) [reply src/state/pools/index.ts:18]: Will check it out, haven't seen it before 👀

File: src/components/Balance.tsx
Comment (Chef-Cheems) [reply src/components/Balance.tsx:11]: I think I tried to pass it just like that but TS couldn't find it... will try again, maybe I did something wrong last time 🤔

File: src/views/Pools/components/PoolTabButtons.tsx
Comment (Chef-Cheems) [reply src/views/Pools/components/PoolTabButtons.tsx:58]: Good to know 👍

File: src/views/Pools/components/PoolsTable/Cells/EarningsCell.tsx
Comment (Chef-Cheems) [reply src/views/Pools/components/PoolsTable/Cells/EarningsCell.tsx:24]: So there is bunch of weird flex ratios here and there, this is mostly fine-tuned in order to show table properly on all screen sizes. Whole numbers are nice but as you resize the window it looks awful at certain resolutions.
Even if there is solution for that that can use whole numbers it would probably require other complex stuff around it.
And non-integer flex values are supported in every browser as far as I know (maybe except IE11) so shouldn't cause any issues 🙂

File: src/views/Pools/components/PoolsTable/PoolsTable.tsx
Comment (Chef-Cheems) [reply src/views/Pools/components/PoolsTable/PoolsTable.tsx:20]: yeah those are not in UIKit, will confirm with Bugs exact shadows and put it there 👍

Comment (Chef-Cheems) [reply src/views/Pools/components/PoolsTable/PoolsTable.tsx:20]: actually talked with Bugs, we're ditching shadows to outlines, will rework this

File: src/views/Pools/components/PoolsTable/Apr.tsx
Comment (ChefHutch) [reply src/views/Pools/components/PoolsTable/Apr.tsx:42]: Yea, and maybe it's time to increase it. We had compounding every block at first and it hasn't slowed down _that_ much. Worth seeing what @Chef-Chungus / Hops thinks about this. TBH the impact on APY between 288 and 28k daily compounds isn't that significant.

File: src/components/Balance.tsx
Comment (ChefHutch) [reply src/components/Balance.tsx:11]: I think if you didn't pass it at all within the `Balance` component Cheems, it may just work? 

`<Text>` accepts `{...props}` and then the `TextProps` should allow an onClick event?

File: src/views/Pools/components/PoolsTable/ActionPanel/ActionPanel.tsx
Comment (ChefHutch) [inline src/views/Pools/components/PoolsTable/ActionPanel/ActionPanel.tsx:166]: Some visual feedback on hover for this icon might be nice to indicate that it's clickable? I'm not sure the Block Balance & text is actually triggering the link? The timer icon is though!

File: src/views/Pools/components/PoolsTable/ActionPanel/ActionPanel.tsx
Comment (ChefHutch) [inline src/views/Pools/components/PoolsTable/ActionPanel/ActionPanel.tsx:136]: Is this sort of thing (not all of it, but a lot of it) the kind of thing you were talking about in terms of having it in helpers, as it's used in a few places?

File: src/views/Pools/components/PoolsTable/ActionPanel/ActionPanel.tsx
Comment (ChefHutch) [inline src/views/Pools/components/PoolsTable/ActionPanel/ActionPanel.tsx:224]: I think this should just be 16px?

File: src/views/Pools/components/PoolsTable/Apr.tsx
Comment (ChefHutch) [inline src/views/Pools/components/PoolsTable/Apr.tsx:39]: It seems a bit of a shame to have stuff like this repeated from the card `AprRow` component rather than creating something that can be shared? It adds a bit of maintenance to make these changes in multiple components going forward. It's pretty likely that these decimals will need tweaking in future.

File: src/views/Pools/components/PoolsTable/ActionPanel/Stake.tsx
Comment (ChefHutch) [inline src/views/Pools/components/PoolsTable/ActionPanel/Stake.tsx:119]: This should maybe be abstracted out of this component & shared with the card ui components rather than duplicating?

File: src/views/Pools/components/PoolsTable/ActionPanel/Harvest.tsx
Comment (ChefHutch) [inline src/views/Pools/components/PoolsTable/ActionPanel/Harvest.tsx:58]: Obviously we do a lot of this in the card too. I feel like all of my recent comments are on this theme - can we be smarter and share the logic between the table / card view where possible?

File: src/views/Pools/components/PoolsTable/Cells/EarningsCell.tsx
Comment (ChefHutch) [inline src/views/Pools/components/PoolsTable/Cells/EarningsCell.tsx:66]: Is this the same as https://github.com/pancakeswap/pancake-frontend/pull/1258/files/b4bf66aee61da2b82999acd9b55da26ec710d8c0#diff-402e4cb9074d844667b90f43f90e042e9bc0d3fa8bdcd9bae57434dd5010373eR55-R58?

File: src/views/Pools/components/HelpButton.tsx
Comment (hachiojidev) [reply src/views/Pools/components/HelpButton.tsx:14]: You know this brings up an issue I have been wrestling with. If you are already extending the style is it better to put the stuff in here to keep the component below clean? I don't know but I have done this before. My guess it is basically the same thing.

File: src/views/Pools/components/PoolsTable/ActionPanel/Harvest.tsx
Comment (Chef-Cheems) [reply src/views/Pools/components/PoolsTable/ActionPanel/Harvest.tsx:58]: Reply to this and couple of other comments about copy-paste.
Yeah, I was kinda trying not to mess with cards initially so I don't end up breaking everything. I'm now refactoring/extracting some common stuff where possible. Not everything though since for example prices are getting moved to redux state by you so there is no point making more abstractions based on hooks.

File: src/views/Pools/components/HelpButton.tsx
Comment (ChefHutch) [reply src/views/Pools/components/HelpButton.tsx:14]: Yea for sure. I'm pro that. Applying styles in two places on the same component will no doubt make debugging a bit trickier, even if the syntax is marginally neater 👍

**Comment by Chef-Cheems** [comment]
@hachiojidev @ChefHutch Updated bunch of stuff here, commits starting from d18177b826ea618725f57115f7cd82539668f815 are new.
In short what happened:
- Using Farm/LP prices now
- Common logic is shared between cards / table
- Sorting by APR & Earned works
- Some minor bug fixes here and there

Preview also updated - https://awesome-raman-0813c6.netlify.app/

File: src/views/Pools/components/HelpButton.tsx
Comment (Chef-Cheems) [reply src/views/Pools/components/HelpButton.tsx:29]: endIcon wouldn't work here since we need to hide the text on small screens, with `endIcon` it's not possible.

File: src/components/Balance.tsx
Comment (Chef-Cheems) [reply src/components/Balance.tsx:11]: update on that - yea, TextProps doesn't allow onClick at all, needs to be explicitly defined here. TextProps inherits only from SpaceProps, TypographyProps, LayoutProps, and they don't contain onClick definition

**Review by hachiojidev — APPROVED** [review]

**Comment by Chef-Cheems** [comment]
Rebased and force-pushed after https://github.com/pancakeswap/pancake-frontend/pull/1348 got merged

File: src/hooks/useApprove.ts
Comment (ChefHutch) [inline src/hooks/useApprove.ts:88]: (following our slack convos)

This `setLastUpdated` call is used to manually trigger a state change that used to cause the `checkApprovalStatus`  function to re-fire (not particularly elegant, I know).

On ln 106, you're instantiating a new `lastUpdated` & `setLastUpdated`, which I don't think will be responsive to `setLastUpdated()` being called here, on ln 88.

The impact for a user is that they have to refresh to trigger the vault approval check. Until they do that - they'll be able to endlessly submit approval transactions.

**Review by ChefHutch — CHANGES_REQUESTED** [review]

File: src/hooks/useApprove.ts
Comment (Chef-Cheems) [reply src/hooks/useApprove.ts:88]: I think I've tested approving/disapproving multiple times after this change and it worked fine. Will double-check it once nodes become responsive.....

Comment (Chef-Cheems) [reply src/hooks/useApprove.ts:88]: Seems to be fine. I think all those other hooks involved in the process somehow "cache" the state 😄 
https://user-images.githubusercontent.com/81824236/119792445-5e5b7000-bede-11eb-8e7d-ec41c67bf620.mp4

Comment (ChefHutch) [reply src/hooks/useApprove.ts:88]: 👍 nice one, ah yea it could be because these are being called within hooks. yolo

**Review by ChefHutch — APPROVED** [review]
Looks great & seems to work well

**Pull request merged**
